go-controller/pkg/ovn/egressip.go

```diff
@@ -366,7 +487,7 @@ func (e *egressIPMode) getGatewayRouterJoinIP(node string, wantsIPv6 bool) (net.
 	} else {
 		err := utilwait.ExponentialBackoff(retry.DefaultRetry, func() (bool, error) {
 			var err error
-			gatewayIPs, err = util.GetNodeLogicalRouterIPs(node)
+			gatewayIPs, _, err = util.GetNodeGatewayRouterNetInfo(node)
```

Comment:
Maybe add a `klog.Errorf` here for err returned by `GetNodeGatewayRouterNetInfo`, the outer err returned by `utilwait` doesn't contain detailed reason why `GetNodeGatewayRouterNetInfo` fails.

Reply:
I am not convinced this is needed, `utilwait.ExponentialBackoff` will return `timed out waiting for the condition` when it reaches the timeout, given that we return and log this in the caller: the logs should already have something like; `"unable to retrieve gateway IP for node: my-node, err: timed out waiting for the condition` which I think is fine. This function is not likely to retry as the gateway router's logical router port should be there since the cluster creation.

Reply:
I was looking at the log in bug: https://bugzilla.redhat.com/show_bug.cgi?id=1881269, where it says something like:
```
02:02:25.555597       1 egressip.go:194] Unable to add pod: test/hello-pod to EgressIP: egressip, err: unable to create logical router policy for status: {compute-0 136.144.52.215}, err: unable to retrieve node's: compute-0 gateway IP, err: timed out waiting for the condition
```
what really failed was inside `GetNodeLogicalRouterIPs` where it didn't get the gateway IP because in new local gateway mode, `lb_force_snat_ip` is not set.

Reply:
OK, understood. I will add it.